contracts/exchangeIssuance/FlashMintWrapped.sol

```diff
@@ -17,220 +17,992 @@
-
-import { IController } from "../interfaces/IController.sol";
-import { IDebtIssuanceModule } from "../interfaces/IDebtIssuanceModule.sol";
-import { ISetToken } from "../interfaces/ISetToken.sol";
-import { IWETH } from "../interfaces/IWETH.sol";
-import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+
+import {IController} from "../interfaces/IController.sol";
+import {IIntegrationRegistry} from "../interfaces/IIntegrationRegistry.sol";
+import {IWrapV2Adapter} from "../interfaces/IWrapV2Adapter.sol";
+import {IDebtIssuanceModule} from "../interfaces/IDebtIssuanceModule.sol";
+import {ISetToken} from "../interfaces/ISetToken.sol";
+import {IWETH} from "../interfaces/IWETH.sol";
+import {IWrapModuleV2} from "../interfaces/IWrapModuleV2.sol";
+import { Withdrawable } from "external/contracts/aaveV2/utils/Withdrawable.sol";
+import {DEXAdapter} from "./DEXAdapter.sol";
 
 /**
  * @title FlashMintWrapped
  *
- * Flash issues SetTokens whose components are purely made up of wrapped tokens.
- * In particular, for issuance, the contract needs to:
- * 1. For all components in a SetToken, swap input token into matching unwrapped component
- * 2. For each unwrapped component, execute the wrapping function
- * 3. Call on issuanceModule to issue tokens
+ * Flash issues SetTokens whose components contain wrapped tokens.
+ *
+ * Compatible with:
+ * IssuanceModules: DebtIssuanceModule, DebtIssuanceModuleV2
+ * WrapAdapters: IWrapV2Adapter
+ *
+ * Supports flash minting for sets that contain both unwrapped and wrapped components.
+ * Does not support debt positions on Set token.
+ * Wrapping / Unwrapping is skipped for a component if ComponentSwapData[component_index].underlyingERC20 address == set component address
+ *
+ * If the set contains both the wrapped and unwrapped version of a token (e.g. DAI and cDAI)
+ * ->  two separate component data points have to be supplied in _swapData and _wrapData
+ * e.g. for issue
+ * Set components at index 0 = DAI; then -> ComponentSwapData[0].underlyingERC20 = DAI; (no wrapping will happen)
+ * Set components at index 1 = cDAI; then -> ComponentSwapData[1].underlyingERC20 = DAI; (wrapping will happen)
  */
-contract FlashMintWrapped is ReentrancyGuard {
-
-    using Address for address payable;
-    using SafeMath for uint256;
-    using PreciseUnitMath for uint256;
-    using SafeERC20 for IERC20;
-    using SafeERC20 for ISetToken;
-
-    struct WrappedCalldata {
-        address target;
-        uint256 quantity;
-        bytes callData;
-    }
+contract FlashMintWrapped is ReentrancyGuard, Withdrawable {
```

Comment:
`Withdrawable` seems to be imported from `external/aavev2/utils` which doesn't make sense as an import for this contract. I think we can either move that out to a shared utils or just use what @ckoopmann did in ExchangeIssuanceZeroEx 
```
    /**
     * Withdraw slippage to selected address
     *
     * @param _tokens    Addresses of tokens to withdraw, specifiy ETH_ADDRESS to withdraw ETH
     * @param _to        Address to send the tokens to
     */
    function withdrawTokens(IERC20[] calldata _tokens, address payable _to) external onlyOwner payable {
        for(uint256 i = 0; i < _tokens.length; i++) {
            if(address(_tokens[i]) == ETH_ADDRESS){
                _to.sendValue(address(this).balance);
            }
            else{
                _tokens[i].safeTransfer(_to, _tokens[i].balanceOf(address(this)));
            }
        }
    }
```

Reply:
done